discord/utils.py

```diff
@@ -900,7 +900,7 @@ def resolve_template(code: Template | str) -> str:
 
 _URL_REGEX = r"(?P<url><[^: >]+:\/[^ >]+>|(?:https?|steam):\/\/[^\s<]+[^<.,:;\"\'\]\s])"
 
-_MARKDOWN_STOCK_REGEX = rf"(?P<markdown>[_\\~|\*`]|{_MARKDOWN_ESCAPE_COMMON})"
+_MARKDOWN_STOCK_REGEX = rf"(?P<markdown>[_\\~|\*`#-]|{_MARKDOWN_ESCAPE_COMMON})"
```

Comment:
I believe the # needs a space to actually render as a header, bullets might be the same
Otherwise this will match channel mentions, and I don't believe it's supposed to

Reply:
The `#` and `-` also need to be at the start of a line to work

Reply:
tbh this is probably only an issue for `#`, if it's an unnecessary escape it will just hide the backslash but for mentions it will break the mention

Reply:
Possibly add the extra regex for the `as_needed` parameter?